traits/observers/_list_change_event.py

```diff
@@ -0,0 +1,72 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Event object for representing mutations to a list.
+"""
+
+# ListChangeEvent is exposed in the public API
+
+
+class ListChangeEvent:
+    """ Event object to represent mutations to a list.
+
+    Attributes
+    ----------
+    trait_list : list
+        The list being mutated.
+    index : int or slice
+        The index used for the mutation.
+    added : list
+        Values added to the list.
+    removed : list
+        Values removed from the list.
+    """
+
+    def __init__(self, *, trait_list, index, removed, added):
```

Comment:
Is there are particular reason you want these to be keyword only?

Reply:
Enforcing keyword-only arguments will help prevent bugs like this one: https://github.com/enthought/traitsui/issues/791

Reply:
I agree that keyword-only arguments make sense here: it makes the calling code clearer as well, particularly with respect to the ordering of `removed` and `added` - someone reading the client code doesn't have to check or remember which way around the arguments go.

The keyword-only restriction is also something that's easy to relax if it becomes problematic. It's harder to add the restriction back in if it's not present to begin with.